src/store/participantsStore.js

```diff
@@ -600,12 +617,27 @@ const actions = {
 
 		try {
 			const response = await request(token)
-			context.dispatch('purgeParticipantsStore', token)
-
 			const hasUserStatuses = !!response.headers['x-nextcloud-has-user-statuses']
 
-			response.data.ocs.data.forEach(participant => {
-				context.dispatch('addParticipant', { token, participant })
+			const currentParticipants = context.state.attendees[token]
+			const newParticipants = response.data.ocs.data
+			for (const attendeeId of Object.keys(Object(currentParticipants))) {
+				if (!newParticipants.some(p => {
+					return p.attendeeId === Number(attendeeId)
+				})) {
```

Comment:
```suggestion
				if (!newParticipants.some(p => p.attendeeId === attendeeId)) {
```

Reply:
`Object.keys` returns keys as strings, so we should have a `Number()` notation. but yes, could be one-line

Reply:
Then you can use `==` (equal value without equal type) instead

Reply:
```js
newParticipants.some(p => {
	return p.attendeeId === Number(attendeeId)
})
```

Also can be replaced by this, saving both Number casting and strict comparison, but in a short way.

```js
p => p.attendeeId === +attendeeId
```

I'd avoid writing multi-line functions in an IF condition.